### PR TITLE
Added zmqpp version 4.2.0

### PR DESCRIPTION
**Pull request by ErniBrown**
* I've checked this [Git style guide](https://0.readthedocs.io/en/latest/git.html). **Yes**
* I've checked this [CMake style guide](https://0.readthedocs.io/en/latest/cmake.html). **Yes**
* My change will work with CMake 3.2 (minimum requirement for Hunter). **Yes**
* I will try to keep this pull request as small as possible and will try not to mix unrelated features. **Yes**

Some help would be appreciated... I've got errors on testing:
- [Windows](https://ci.appveyor.com/project/ErniBrown/hunter/builds/23594350): I'm not used to windows compiling, but I think almost every test that failed was trying to build dependency zeromq, not zmqpp; what should I do in this case?
- [Linux&Mac](https://travis-ci.com/ErniBrown/hunter/builds/107050397): I've got two errors:
  - One is from clang, it seems to me that is failing on a couple of warning. What should I do, disable warning during compile? I do not want to fix the error in the code
  - The other one is in 'IOS' compilying, and i never compiled for this system, so I don't understand what's happening

### CODE REVIEW

**Comment by ErniBrown** [comment]
@ruslo Do you receive notifications about draft PR?

**Comment by ruslo** [comment]
> Some help would be appreciated... I've got errors on testing:

You are not running tests correctly. You should fetch CI configs from `pkg.zmqpp` branch. There is only single test in config:
- https://github.com/ingenue/hunter/blob/pkg.zmqpp/.travis.yml#L103-L110

File: cmake/projects/ZMQPP/hunter.cmake [hunk 1]
```diff
@@ -27,6 +27,17 @@ hunter_add_version(
   fdc3a0d5a4e907c79709ec2fc3fcf4105ff1b91a
   )
 
+hunter_add_version(
+  PACKAGE_NAME
+  ZMQPP
+  VERSION
+  "4.2.0"
```
Comment (ruslo) [inline cmake/projects/ZMQPP/hunter.cmake]: Please keep Git tag and `VERSION` in a consistent state:
- https://docs.hunter.sh/en/latest/creating-new/create/cmake.html#consistency

**Comment by ErniBrown** [comment]
I'm sorry, but something is not clear to me: in wich branch or repository should i change the .travis and appveyor files? Should i open a pull request for them on ingenue/hunter?
I followed the [ci-testing](https://docs.hunter.sh/en/latest/creating-new/create/cmake.html#ci-testing) guide, but I was not sure about this.

**Comment by ruslo** [comment]
Testing:
- https://ci.appveyor.com/project/ingenue/hunter/builds/23628400
- https://travis-ci.org/ingenue/hunter/builds/516222152?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
- https://ci.appveyor.com/project/ingenue/hunter/builds/23628692
- https://travis-ci.org/ingenue/hunter/builds/516227362?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.23.151

**Pull request merged**